### PR TITLE
[LI-HOTFIX] Add debugger to print resolved IPs, and assign clientId to ProducerConfig

### PR DESCRIPTION
**Pull request by fluffywei**
Added debugger to print resolved IPs; assigned the self generated producer clientId to ProducerConfig when users haven't provided one.

EXIT_CRITERIA = When gets to the root of how the client’s DNS resolver gets it’s IPs in a kafka clients test in EI cluster

More detailed description of your change,
if necessary. The PR title and PR message become
the squashed commit message, so use a separate
comment to ping reviewers.

Summary of testing strategy (including rationale)
for the feature or bug fix. Unit and/or integration
tests are expected for any behaviour change and
system tests should be considered for larger changes.

Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: clients/src/main/java/org/apache/kafka/clients/ClientUtils.java [hunk 1]
```diff
@@ -111,6 +111,10 @@ public static ChannelBuilder createChannelBuilder(AbstractConfig config, Time ti
 
     static List<InetAddress> resolve(String host, ClientDnsLookup clientDnsLookup) throws UnknownHostException {
         InetAddress[] addresses = InetAddress.getAllByName(host);
+        log.debug("Resolved {} and got addresses: ", host);
+        for (InetAddress address: addresses) {
+            log.debug(address.getHostAddress() + ";");
```
Comment (kehuum) [inline clients/src/main/java/org/apache/kafka/clients/ClientUtils.java:116]: this getHostAddress prints the actual resolved host name, not the bootstrap dns disco address, right?

File: clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java [hunk 2]
```diff
@@ -331,13 +331,20 @@ public KafkaProducer(Properties properties, Serializer<K> keySerializer, Seriali
...
             String transactionalId = userProvidedConfigs.containsKey(ProducerConfig.TRANSACTIONAL_ID_CONFIG) ?
                     (String) userProvidedConfigs.get(ProducerConfig.TRANSACTIONAL_ID_CONFIG) : null;
 
-            this.clientId = buildClientId(config.getString(ProducerConfig.CLIENT_ID_CONFIG), transactionalId);
+            String configuredClientId = config.getString(ProducerConfig.CLIENT_ID_CONFIG);
+
+            this.clientId = buildClientId(configuredClientId, transactionalId);
+
+            if (configuredClientId.isEmpty()) {
+                config.originals().put(ProducerConfig.CLIENT_ID_CONFIG, this.clientId);
```
Comment (kehuum) [inline clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java:344]: what's the usage for this? If user didn't supply a client.id, the printed configs will have client.id as empty field?

File: clients/src/main/java/org/apache/kafka/clients/ClientUtils.java [hunk 3]
```diff
@@ -111,6 +111,10 @@ public static ChannelBuilder createChannelBuilder(AbstractConfig config, Time ti
```
Comment (kehuum) [inline clients/src/main/java/org/apache/kafka/clients/ClientUtils.java:111]: nit: you'll need to follow the pr format for kafka repos, similar to https://github.com/linkedin/kafka/pull/329

File: clients/src/main/java/org/apache/kafka/clients/ClientUtils.java (hunk 1 above)
Comment (fluffywei) [reply clients/src/main/java/org/apache/kafka/clients/ClientUtils.java:116]: getHostAddress returns the IP address string in textual presentation.

File: clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java (hunk 2 above)
Comment (fluffywei) [reply clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java:344]: yes, when users didn't config client.id, kafka clients will self generate it and the ProducerConfig values result will be: client.id =

File: clients/src/main/java/org/apache/kafka/clients/ClientUtils.java (hunk 3 above)
Comment (fluffywei) [reply clients/src/main/java/org/apache/kafka/clients/ClientUtils.java:111]: Thanks, modified the format. Lemme know if further adjustment is needed.

**Review by kehuum — APPROVED** [review]

**Pull request merged**